gammapy/catalog/gammacat.py

```diff
@@ -54,25 +54,246 @@ class SourceCatalogObjectGammaCat(SourceCatalogObject):
+
+        if(spec == 'ecpl'):
+
+            unit = 'cm-2 s-1 TeV-1'
+            fmt = '{:<15s} : {:.3} +- {:.3} {} (statistical)\n'
+            args = ('norm', d['spec_ecpl_norm'].value, d['spec_ecpl_norm_err'].value, unit)
+            ss += fmt.format(*args)
+            fmt = '{:<15s}   {:.3} +- {:.3} {} (systematic)\n'
+            args = ('', d['spec_ecpl_norm'].value, d['spec_ecpl_norm_err_sys'].value, unit)
+            ss += fmt.format(*args)
+
+            fmt = '{:<15s} : {:.3} +- {:.3} (statistical)\n'
+            args = ('index', d['spec_ecpl_index'], d['spec_ecpl_index_err'])
+            ss += fmt.format(*args)
+            fmt = '{:<15s}   {:.3} +- {:.3} (systematic)\n'
+            args = ('', d['spec_ecpl_index'], d['spec_ecpl_index_err_sys'])
+            ss += fmt.format(*args)
+
+            unit = 'TeV'
+            fmt = '{:<15s} : {:.3} +- {:.3} {} (statistical)\n'
+            args = ('e_cut', d['spec_ecpl_e_cut'].value, d['spec_ecpl_e_cut_err'].value, unit)
+            ss += fmt.format(*args)
+            fmt = '{:<15s}   {:.3} +- {:.3} {} (systematic)\n'
+            args = ('', d['spec_ecpl_e_cut'].value, d['spec_ecpl_e_cut_err_sys'].value, unit)
+            ss += fmt.format(*args)
+
+            ss += '{:<15s} : {:.3}\n'.format('reference', d['spec_ecpl_e_ref'])
+
+        ss += '\n{:<20s} : {:.3}\n'.format('energy range min', d['spec_erange_min'])
+        ss += '{:<20s} : {:.3}\n'.format('energy range max', d['spec_erange_max'])
+        ss += '{:<20s} : {:.3}\n'.format('theta', d['spec_theta'])
+
+        ss = '\n\nDerived fluxes:\n'
+
+        unit = 'cm-2 s-1 TeV-1'
+        fmt = '{:<30s} : {:.3} +- {:.3} {} (statistical)\n'
+        args = ('Spectral model norm (1 TeV)', d['spec_dnde_1TeV'].value, d['spec_dnde_1TeV_err'].value, unit)
+        ss += fmt.format(*args)
+
+        unit = 'cm-2 s-1'
+        fmt = '{:<30s} : {:.3} +- {:.3} {} (statistical)\n'
+        args = ('Integrated flux (<1 TeV)', d['spec_flux_1TeV'].value, d['spec_flux_1TeV_err'].value, unit)
+        ss += fmt.format(*args)
+
+        unit = '(crab units)'
+        fmt = '{:<30s} : {:.3} +- {:.3} {}\n'
+        args = ('Integrated flux (<1 TeV)', d['spec_flux_1TeV_crab'], d['spec_flux_1TeV_crab_err'], unit)
+        ss += fmt.format(*args)
+
+        unit = 'erg cm-2 s-1'
+        fmt = '{:<30s} : {:.3} +- {:.3} {} (statistical)\n'
+        args = ('Integrated flux (1-10 TeV)', d['spec_eflux_1TeV_10TeV'].value, d['spec_eflux_1TeV_10TeV_err'].value, unit)
+        ss += fmt.format(*args)
+
+        return ss
+
+
+    def _info_spectral_points(self):
+        """Print spectral points info."""
+        d = self.data
+        ss = '\n*** Spectral points ***\n'
+        ss += '{:<20s} : {}\n'.format('SED reference id', d['sed_reference_id'])
+        ss += '{:<20s} : {}\n'.format('# spectral points', d['sed_n_points'])
+        ss += '{:<20s} : {}\n\n'.format('# upper limits', d['sed_n_ul'])
+
+        t = self._flux_points_table_formatted
```

Comment:
I think there are sources without spectral points, where this will raise an error?
If so, I think the whole section should be in an if or try statement, and if there are no spectral points, just print "no spectral points available for this source" instead of erroring out.

To test all sources you should simply be able to do this:
```
for source in catalog:
    print(source.name)
    print(source)
```